### PR TITLE
Merge custom directives with federated schema

### PR DESCRIPTION
**Pull request by padenj**

### CODE REVIEW

**Comment by padenj** [comment]
Workaround for issue #40

**Comment by kzamer** [comment]
There's a need on our team to have the @authorize directive added to queries and mutations.
Really interested in this! 👍

**Comment by tlivings** [comment]
Can you add a test for this?

File: lib/index.js [hunk 1]
```diff
@@ -112,6 +112,19 @@ class GraphQLComponent {
...
+  makeFederatedSchemaWithDirectives({typeDefs, resolvers, schemaDirectives}) {
+    const federatedSchema = buildFederatedSchema({
+      typeDefs,
+      resolvers
+    });
+
+    return mergeSchemas({
+      schemas: [federatedSchema],
+      schemaDirectives,
+      mergeDirectives: true
```
Comment (brianjquinn) [inline lib/index.js]: to get this working in the unit tests, I had to add mergeDirectives: true here.

**Pull request merged**